src/utils.ts

```diff
@@ -41,13 +42,24 @@ export function getEncodingCharset(charset = kDefaultEncodingCharset): BufferEnc
  * If the response as a content type equal to 'application/json' we automatically parse it with JSON.parse().
  */
 export async function parseUndiciResponse<T>(response: Dispatcher.ResponseData): Promise<T | string> {
-  const body = await response.body.text();
   const contentTypeHeader = response.headers["content-type"] as string | undefined;
-  const { type } = contentType.parse(
+  const { type, parameters } = contentType.parse(
     contentTypeHeader ?? kDefaultMimeType
   );
 
+  let body;
   try {
+    if (response.headers["content-encoding"] === "gzip") {
+      const buf: any[] = [];
+      for await (const data of response.body) {
+        buf.push(data);
+      }
+      body = gunzipSync(Buffer.concat(buf)).toString(getEncodingCharset(parameters.charset));
```

Comment:
We should use async version no?

Reply:
:thumbsup: I used 'promisify' to make 'zlib.gunzip' async